numpy/lib/tests/test_mixins.py

```diff
@@ -56,11 +56,47 @@ def __repr__(self):
         return '%s(%r)' % (type(self).__name__, self.value)
 
 
+def wrap_array_like(result):
+    if type(result) is tuple:
+        return tuple(ArrayLike(r) for r in result)
+    else:
+        return ArrayLike(result)
+
+
 def _assert_equal_type_and_value(result, expected, err_msg=None):
     assert_equal(type(result), type(expected), err_msg=err_msg)
-    assert_equal(result.value, expected.value, err_msg=err_msg)
-    assert_equal(getattr(result.value, 'dtype', None),
-                 getattr(expected.value, 'dtype', None), err_msg=err_msg)
+    if isinstance(result, tuple):
+        assert_equal(len(result), len(expected), err_msg=err_msg)
+        for result_item, expected_item in zip(result, expected):
+            _assert_equal_type_and_value(result_item, expected_item, err_msg)
+    else:
+        assert_equal(result.value, expected.value, err_msg=err_msg)
+        assert_equal(getattr(result.value, 'dtype', None),
+                     getattr(expected.value, 'dtype', None), err_msg=err_msg)
+
+
+_ALL_BINARY_OPERATORS = [
+    operator.lt,
+    operator.le,
+    operator.eq,
+    operator.ne,
+    operator.gt,
+    operator.ge,
+    operator.add,
+    operator.sub,
+    operator.mul,
+    operator.truediv,
+    operator.floordiv,
+    # TODO: test div on Python 2, only
+    operator.mod,
+    divmod,
+    pow,
```

Comment:
Good catch. In particular, `operator.pow` has two arguments, but `pow` has three